src/main/java/net/dv8tion/jda/api/entities/Widget.java

```diff
@@ -0,0 +1,806 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.dv8tion.jda.api.entities;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import gnu.trove.map.TLongObjectMap;
+import gnu.trove.map.hash.TLongObjectHashMap;
+import net.dv8tion.jda.api.OnlineStatus;
+import net.dv8tion.jda.api.utils.ImageProxy;
+import net.dv8tion.jda.api.utils.MiscUtil;
+import net.dv8tion.jda.api.utils.data.DataArray;
+import net.dv8tion.jda.api.utils.data.DataObject;
+import net.dv8tion.jda.internal.entities.EntityBuilder;
+
+public class Widget implements ISnowflake
+{
+    private final boolean isAvailable;
+    private final long id;
+    private final String name;
+    private final String invite;
+    private final TLongObjectMap<VoiceChannel> channels;
+    private final TLongObjectMap<Member> members;
+    
+    /**
+     * Constructs an unavailable Widget
+     */
+    public Widget(long guildId)
```

Comment:
The constructors probably should not be exposed

Reply:
We might want to consider making Widget an interface and moving the underlying implementation into the `internal` classes via `WidgetImpl`